client/echo_client.go

```diff
@@ -213,33 +240,33 @@ func NewEchoClient(ctx context.Context, opts ...option.ClientOption) (*EchoClien
 
 	c.locationsClient = locationpb.NewLocationsClient(connPool)
 
-	return c, nil
+	return &EchoClient{internalEchoClient: c, CallOptions: callOpts}, nil
```

Comment:
Let's be careful here: `&(EchoClient.CallOptions) != &callOptions`, right? After this line, we could say `EchoClient.CallOptions = altCallOptions` and then have `*(c.callOpts) != altCallOptions`, right?

I think you want 
```
echoClient = &EchoClient{internalEchoClient: c, CallOptions: callOpts}
c.CallOptions = &EchoClient.CallOptions
return echoEclient
```
Feel free to check me.
```